### PR TITLE
[Design Pattern] Factory Pattern - Car #52

### PR DESCRIPTION
**Pull request by frosse**

### CODE REVIEW

**Comment by fengyuanyang** [comment]
sorry for the build fail , could you pull and merge from master again , i just fixed the fail issues on coverage rates

**Comment by januslinhc** [comment]
> sorry for the build fail , could you pull and merge from master again , i just fixed the fail issues on coverage rates

I think this program is to learn TDD right? if contributors follow TDD practice, the code coverage should be near 100%. If less then that, it means it isnt follow TDD.

**Comment by fengyuanyang** [comment]
> > sorry for the build fail , could you pull and merge from master again , i just fixed the fail issues on coverage rates
> 
> I think this project is to learn TDD right? if contributors follow TDD practice, the code coverage should be close to 100%. If less then that, it means the test cases are not fully covered the code you have implemented (e.g. it isnt follow TDD - test driven development- write test cases first!).

Yes, we enabled the test coverage check recently , but we didn't check the pull request before this change. 
I merged one of the pr yesterday, it fail to pass the coverage check

**Comment by januslinhc** [comment]
> > > sorry for the build fail , could you pull and merge from master again , i just fixed the fail issues on coverage rates
> > 
> > 
> > I think this project is to learn TDD right? if contributors follow TDD practice, the code coverage should be close to 100%. If less then that, it means the test cases are not fully covered the code you have implemented (e.g. it isnt follow TDD - test driven development- write test cases first!).
> 
> Yes, we enabled the test coverage check recently , but we didn't check the pull request before this change.
> I merged one of the pr yesterday, it fail to pass the coverage check

I mean if you follow TDD, you should write the test cases first, and then you write code to make them pass! In other words, redundant and uncovered code is not expected.

**Comment by frosse** [comment]
Rebased from main and check is ok now.

File: src/main/java/com/ordestiny/tdd/designpattern/factorypattern/car/CarType.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.ordestiny.tdd.designpattern.factorypattern.car;
+
+import java.util.function.Supplier;
+
+public enum CarType {
+    BENZ(Benz::new), BMW(Bmw::new), TOYOTA(Toyota::new), JAGUAR(Jaguar::new);
```
Comment (fengyuanyang) [inline src/main/java/com/ordestiny/tdd/designpattern/factorypattern/car/CarType.java:6]: nice code

**Pull request merged**